test/unit/metadata_test.py

```diff
@@ -88,3 +96,67 @@ def test_get_sorted_resources() -> None:
     assert last_resource_name.startswith(
         "_out"
     ), f"{last_resource_name} is the last resource. Expected a resource with the prefix '_out'"
+
+
+def test_resource_descriptors_valid():
+    # just make sure these validate properly
+    descriptors = {
+        name: PudlResourceDescriptor.model_validate(desc)
+        for name, desc in RESOURCE_METADATA.items()
+    }
+    assert len(descriptors) > 0
```

Comment:
The main failure case of this would be "the test hits a validation error." But, in case we really goof something up and `RESOURCE_METADATA.items()` is empty, this test would succeed... so I added some assertion that we really expect to be true all the time.